includes/api/save-abandonment-data.php

```diff
@@ -291,6 +298,433 @@ function saveWooCartAbandonmentRecoveryData($razorpayData)
+          VALUES ( %s, %s, %s, %s, %s, %s, %0.2f, %s, %s, %s, %s)",
+          array(
+              'name'			=> sanitize_text_field( $user_data['name'] ),
+              'surname'		    => sanitize_text_field( $user_data['surname'] ),
+              'email'			=> sanitize_email( $user_data['email'] ),
+              'phone'			=> filter_var( $user_data['phone'], FILTER_SANITIZE_NUMBER_INT),
+              'location'		=> sanitize_text_field( serialize( $user_data['location'] ) ),
+              'products'		=> serialize($cart['product_array']),
+              'total'			=> sanitize_text_field( $cart['cart_total'] ),
+              'currency'		=> sanitize_text_field( $cart['cart_currency'] ),
+              'time'			=> sanitize_text_field($cart['current_time']),
+              'session_id'	    => sanitize_text_field($cart['session_id']),
+              'other_fields'	=> sanitize_text_field(serialize($user_data['other_fields']))
+          )
+      )
+  );
+
+   increaseRecoverableCartCountCB();
+   setCartBountySession($session_id);
+}
+  $updated_rows = $wpdb->query(
+    $wpdb->prepare(
+        "UPDATE $cart_table
+        SET name = %s,
+        surname = %s,
+        email = %s,
+        phone = %s,
+        location = %s,
+        other_fields = '$other_fields'
+        WHERE session_id = %s and type=0",
+        sanitize_text_field( $user_data['name'] ),
+        sanitize_text_field( $user_data['surname'] ),
+        sanitize_email( $user_data['email'] ),
+        filter_var( $user_data['phone'], FILTER_SANITIZE_NUMBER_INT),
+        sanitize_text_field( serialize( $user_data['location'] ) ),
+        $session_id
+    )
+);
+
+    deleteDuplicateCarts( $session_id, $updated_rows,$cart_table);
+    setCartBountySession($session_id);
+    $response['status']    = true;
+    $response['message']   = 'Data successfully inserted for CartBounty plugin';
+    $statusCode            = 200;
+
+    update_post_meta($session_id,'FromEmail',"Y");
+    WC()->session->set( 'cartbounty_from_link', true );
+    
+    $result['response']    = $response;
+    $result['status_code'] = $statusCode;
+    return $result;
+}
+
+function getSessionID($order_id){
+  $session_id = WC()->session->get_customer_id();
+  $order      = wc_get_order( $order_id );
+  $user_id    = $order->get_user_id();
+  
+if($user_id != 0 or $user_id != null){  //Used to check whether user is logged in
+    $session_id=$user_id;
+  }else{
+        $session_id = WC()->session->get( 'cartbounty_session_id' );
+        if( empty( $session_id ) ){ //If session value does not exist - set one now
+          $session_id = WC()->session->get_customer_id(); //Retrieving customer ID from WooCommerce sessions variable
+         }
+       if( WC()->session->get( 'cartbounty_from_link' ) && WC()->session->get( 'cartbounty_session_id' ) ){
+          $session_id = WC()->session->get( 'cartbounty_session_id' );
+         }
+  }
+  return $session_id;
+}
+
+function handleOrder( $order_id ){
```

Comment:
@angbpy Function name should be `handleCBRecoveredOrder`

Reply:
Done and updated